openbas-api/src/main/java/io/openbas/executors/crowdstrike/service/CrowdStrikeExecutorContextService.java

```diff
@@ -55,48 +57,178 @@ public void launchExecutorSubprocess(
             .orElseThrow(
                 () -> new UnsupportedOperationException("Inject does not have a contract"));
 
-    String scriptName;
-    String implantLocation;
-    switch (platform) {
-      case Windows -> {
-        scriptName = this.crowdStrikeExecutorConfig.getWindowsScriptName();
-        implantLocation =
-            "$location="
-                + IMPLANT_LOCATION_WINDOWS
-                + ExecutorHelper.IMPLANT_BASE_NAME
-                + UUID.randomUUID()
-                + "\";md $location -ea 0;[Environment]::CurrentDirectory";
-      }
-      case Linux, MacOS -> {
-        scriptName = this.crowdStrikeExecutorConfig.getUnixScriptName();
-        implantLocation =
-            "location="
-                + IMPLANT_LOCATION_UNIX
-                + ExecutorHelper.IMPLANT_BASE_NAME
-                + UUID.randomUUID()
-                + ";mkdir -p $location;filename=";
+    manageWithoutPlatformAgents(agents, injectStatus);
+    List<CrowdStrikeAction> actions = new ArrayList<>();
+    // Set implant script for Windows CS agents
+    actions.addAll(
+        getWindowsActions(
+            agents.stream()
+                .filter(
+                    agent ->
+                        ((Endpoint) agent.getAsset())
+                            .getPlatform()
+                            .equals(Endpoint.PLATFORM_TYPE.Windows))
+                .toList(),
+            injector,
+            inject.getId()));
+    // Set implant script for Linux CS agents
+    actions.addAll(
+        setLinuxActions(
+            agents.stream()
+                .filter(
+                    agent ->
+                        ((Endpoint) agent.getAsset())
+                            .getPlatform()
+                            .equals(Endpoint.PLATFORM_TYPE.Linux))
+                .toList(),
+            injector,
+            inject.getId()));
+    // Set implant script for Mac CS agents
+    actions.addAll(
+        setMacActions(
+            agents.stream()
+                .filter(
+                    agent ->
+                        ((Endpoint) agent.getAsset())
+                            .getPlatform()
+                            .equals(Endpoint.PLATFORM_TYPE.MacOS))
+                .toList(),
+            injector,
+            inject.getId()));
+    // Launch payloads with CS API
+    executeActions(actions);
+  }
+
+  private void manageWithoutPlatformAgents(List<Agent> agents, InjectStatus injectStatus) {
+    List<Agent> withoutPlatformAgents =
+        agents.stream()
+            .filter(
+                agent ->
+                    ((Endpoint) agent.getAsset()).getPlatform() == null
+                        || ((Endpoint) agent.getAsset()).getPlatform()
+                            == Endpoint.PLATFORM_TYPE.Unknown
+                        || ((Endpoint) agent.getAsset()).getArch() == null)
+            .toList();
+    agents.removeAll(withoutPlatformAgents);
```

Comment:
this is dangerously placed. It modifies a list given as parameter which could create errors down the line in the future (people using this list not knowing it has been modified. Maybe prefer returning a new list instead and calling your method with agents = manageWithoutPlatformAgents(agents...) ?